openldap/2.6/debian-11/rootfs/opt/bitnami/scripts/libopenldap.sh

```diff
@@ -0,0 +1,1045 @@
+# Configure LDAP credentials for admin user
+# Globals:
+#   LDAP_*
+# Arguments:
+#   None
+# Returns:
+#   None
+#########################
+ldap_admin_credentials() {
+    info "Configure LDAP credentials for admin user"
+    cat > "${LDAP_SHARE_DIR}/admin.ldif" << EOF
+dn: olcDatabase={2}mdb,cn=config
+changetype: modify
+replace: olcSuffix
+olcSuffix: ${LDAP_ROOT}
+
+dn: olcDatabase={2}mdb,cn=config
+changetype: modify
+replace: olcRootDN
+olcRootDN: ${LDAP_ADMIN_DN}
+
+dn: olcDatabase={2}mdb,cn=config
+changeType: modify
+add: olcRootPW
+olcRootPW: ${LDAP_ENCRYPTED_ADMIN_PASSWORD}
+
+dn: olcDatabase={1}monitor,cn=config
+changetype: modify
+replace: olcAccess
+olcAccess: {0}to * by dn.base="gidNumber=$(id -g)+uidNumber=$(id -u),cn=peercred,cn=external, cn=auth" read by dn.base="${LDAP_ADMIN_DN}" read by * none
+EOF
+    if is_boolean_yes "$LDAP_CONFIG_ADMIN_ENABLED"; then
+        cat >> "${LDAP_SHARE_DIR}/admin.ldif" << EOF
+
+dn: olcDatabase={0}config,cn=config
+changetype: modify
+add: olcRootDN
+olcRootDN: $LDAP_CONFIG_ADMIN_DN
+
+dn: olcDatabase={0}config,cn=config
+changetype: modify
+add: olcRootPW
+olcRootPW: $LDAP_ENCRYPTED_CONFIG_ADMIN_PASSWORD
+EOF
+    fi
+    debug_execute ldapmodify "${slapd_debug_args[@]}" -Y EXTERNAL -H "$LDAP_LDAPI_URI" -f "${LDAP_SHARE_DIR}/admin.ldif"
+}
+
+########################
+# Disable LDAP anonymous bindings
+# Globals:
+#   LDAP_*
+# Arguments:
+#   None
+# Returns:
+#   None
+#########################
+ldap_disable_anon_binding() {
+    info "Disable LDAP anonymous binding"
+    cat > "${LDAP_SHARE_DIR}/disable_anon_bind.ldif" << EOF
+dn: cn=config
+changetype: modify
+add: olcDisallows
+olcDisallows: bind_anon
+
+dn: cn=config
+changetype: modify
+add: olcRequires
+olcRequires: authc
+EOF
+    debug_execute ldapmodify "${slapd_debug_args[@]}" -Y EXTERNAL -H "$LDAP_LDAPI_URI" -f "${LDAP_SHARE_DIR}/disable_anon_bind.ldif"
+}
+
+########################
+# Add LDAP schemas
+# Globals:
+#   LDAP_*
+# Arguments:
+#   None
+# Returns
+#   None
+#########################
+ldap_add_schemas() {
+    info "Adding LDAP extra schemas ${LDAP_EXTRA_SCHEMAS} ..."
+    read -r -a schemas <<< "$(tr ',;' ' ' <<< "${LDAP_EXTRA_SCHEMAS}")"
+    for schema in "${schemas[@]}"; do
+        info "\t${LDAP_CONF_DIR}/schema/${schema}.ldif"
+        if [ ! -f "${LDAP_CONF_DIR}/schema/${schema}.ldif" ]; then
+            error "Extra schema ${schema} does not exist at ${LDAP_CONF_DIR}/schema/${schema}.ldif"
+        fi
+        debug_execute ldapadd "${slapd_debug_args[@]}" -Y EXTERNAL -H "$LDAP_LDAPI_URI" -f "${LDAP_CONF_DIR}/schema/${schema}.ldif"
+    done
+}
+
+########################
+# Add custom schema
+# Globals:
+#   LDAP_*
+# Arguments:
+#   None
+# Returns
+#   None
+#########################
+ldap_add_custom_schema() {
+    info "Adding custom Schema from ${LDAP_CUSTOM_SCHEMA_FILE} ..."
+    debug_execute slapadd "${slapd_debug_args[@]}" -F "$LDAP_ONLINE_CONF_DIR" -n 0 -l  "$LDAP_CUSTOM_SCHEMA_FILE"
```

Comment:
If you use ldapadd, you can load them at runtime (assuming slapd is actually running)